schema_based_on_diagram.sql

```diff
@@ -0,0 +1,54 @@
+CREATE TABLE "specializations"(
+    "species_Id" INTEGER NOT NULL,
+    "vet_id" INTEGER NOT NULL
+);
+CREATE TABLE "vets"(
+    "id" BIGINT NOT NULL,
+    "name" VARCHAR(255) NOT NULL,
+    "age" INTEGER NOT NULL,
+    "date_of_graduation" DATE NOT NULL
+);
+ALTER TABLE
+    "vets" ADD PRIMARY KEY("id");
+CREATE TABLE "visits"(
+    "animal_id" INTEGER NOT NULL,
+    "vet_id" INTEGER NOT NULL,
+    "date_of_visit" DATE NOT NULL
+);
+CREATE TABLE "animals"(
+    "id" INTEGER NOT NULL,
+    "name" VARCHAR(255) NOT NULL,
+    "date_of_birth" DATE NOT NULL,
+    "escape_attempts" INTEGER NOT NULL,
+    "neutered" BOOLEAN NOT NULL,
+    "weight_kg" DOUBLE PRECISION NOT NULL,
+    "species_id" INTEGER NOT NULL,
+    "owner_id" INTEGER NOT NULL
+);
+ALTER TABLE
+    "animals" ADD PRIMARY KEY("id");
+CREATE TABLE "species"(
+    "id" BIGINT NOT NULL,
+    "name" VARCHAR(255) NOT NULL
+);
+ALTER TABLE
+    "species" ADD PRIMARY KEY("id");
+CREATE TABLE "owners"(
+    "id" INTEGER NOT NULL,
+    "full_name" VARCHAR(255) NOT NULL,
+    "age" INTEGER NOT NULL
+);
+ALTER TABLE
+    "owners" ADD PRIMARY KEY("id");
+ALTER TABLE
+    "specializations" ADD CONSTRAINT "specializations_species_id_foreign" FOREIGN KEY("species_Id") REFERENCES "species"("id");
+ALTER TABLE
+    "visits" ADD CONSTRAINT "visits_vet_id_foreign" FOREIGN KEY("vet_id") REFERENCES "vets"("id");
+ALTER TABLE
+    "specializations" ADD CONSTRAINT "specializations_vet_id_foreign" FOREIGN KEY("vet_id") REFERENCES "vets"("id");
+ALTER TABLE
+    "animals" ADD CONSTRAINT "animals_species_id_foreign" FOREIGN KEY("species_id") REFERENCES "species"("id");
+ALTER TABLE
+    "visits" ADD CONSTRAINT "visits_animal_id_foreign" FOREIGN KEY("animal_id") REFERENCES "animals"("id");
+ALTER TABLE
+    "animals" ADD CONSTRAINT "animals_owner_id_foreign" FOREIGN KEY("owner_id") REFERENCES "owners"("id");
```

Comment:
- [ ] Great work on your progress so far! 👏 However, I'd like to point out an important detail regarding the project requirements. It appears that the diagram you've used as a reference for your project might not align with the specified requirements. To ensure the project's success, I recommend thoroughly reviewing the project requirements document and adhering to the provided instructions. Here is the correct diagram you should be using, as outlined in the project requirements:
![clinic_diagram](https://github.com/mohammadnewton/vet_clinic/assets/107192076/c7a708eb-6b1d-443d-8170-402b5b958212)

Please make the necessary adjustments to your implementation based on this diagram to ensure that your project aligns perfectly with the requirements.

Happy coding! 😊🚀🍀